pkg/handler/processor/ite6/ite6.go

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ite6
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/in-toto/in-toto-golang/in_toto"
+)
+
+type ite6Type string
+
+const (
+	slsaPredicateType ite6Type = "https://slsa.dev/provenance/v0.2"
+)
+
+type ITE6Processor struct {
+}
+
+// ValidateSchema ensures that the document blob can be parsed into a valid data structure
+func (e *ITE6Processor) ValidateSchema(i *processor.Document) error {
+	if i.Type != processor.DocumentITE6 {
+		return fmt.Errorf("expected document type: %v, actual document type: %v", processor.DocumentITE6, i.Type)
+	}
+
+	_, err := parseStatement(i.Blob)
+
+	return err
+}
+
+// Unpack takes in the document and tries to unpack the provenance.
+// if the predicate is of SLSA type the predicate is stored in the blob
+func (e *ITE6Processor) Unpack(i *processor.Document) ([]*processor.Document, error) {
+	if i.Type != processor.DocumentITE6 {
+		return nil, fmt.Errorf("expected document type: %v, actual document type: %v", processor.DocumentITE6, i.Type)
+	}
+
+	statement, err := parseStatement(i.Blob)
+	if err != nil {
+		return nil, err
+	}
+	var doc *processor.Document
+	predicatePayload, err := getPredicate(statement)
+	if err != nil {
+		return nil, err
+	}
+	switch pt := statement.PredicateType; pt {
+	case string(slsaPredicateType):
+		doc = &processor.Document{
+			Blob:              predicatePayload,
+			Type:              processor.DocumentSLSA,
```

Comment:
hmm, i think this is a point of discussion, is `DocumentSLSA` the entire intoto JSON or the contents of the predicate? This does make sense, but also SLSA provenance definition is based on the entire document as well (https://slsa.dev/provenance/v0.2).

Reply:
Right. Should it be the whole provenance or just the predicate was the question.

Reply:
I have some thoughts around this. Hard to type out. In the graph it might make sense because we establish the relationship through edges anyway.

Reply:
Also FWIW, folks have recognized that the SLSA stuff is a bit backwards. We should be separating provenance from the in-toto wrapper, otherwise we end up with situations where we make it difficult to have composite types.

Reply:
i think the tricky part here though is that some of the information is missing as as a standalone predicate payload - i.e. the subject will not be available. Which makes me want to tend towards having SLSA be the entire ITE6, for ease of implementation initially.

Reply:
Isn't that why the doc tree was created. This would be similar to the DSSE processor where we just store the payload. Losing the signature information.

Reply:
I think the main motivation was more for trust validation to simplify the processor. 

This makes the ingestor less modular since a SLSA document ingestor now needs to somehow feed information from the parent node - which may or may not exist. Then some kind of map would need to be passed down but figuring out if a field was from the parent or previous ancestors would need to be considered as well. 

I guess the question would be if a document (minus trust validation) should be self contained. Like SBOMs describe the subject within itself through an identifier. Slsa should also duplicate that information within its predicate payload?